src/coprocessor/statistics/analyze.rs

```diff
@@ -545,23 +545,31 @@ impl BaseRowSampleCollector {
                 self.null_count[col_len + i] += 1;
                 continue;
             }
-            // Use a in place murmur3 to replace this memory copy.
-            for j in offsets {
-                if columns_info[*j as usize].as_accessor().is_string_like() {
-                    self.row_buf
-                        .extend_from_slice(&collation_keys_val[*j as usize]);
+            if offsets.len() == 1 {
+                let offset = offsets[0] as usize;
+                if columns_info[offset].as_accessor().is_string_like() {
+                    self.fm_sketches[col_len + i].insert(&collation_keys_val[offset]);
                 } else {
-                    self.row_buf.extend_from_slice(&columns_val[*j as usize]);
+                    self.fm_sketches[col_len + i].insert(&columns_val[offset]);
+                }
+            } else {
+                let mut hasher = Hasher128::with_seed(0);
+                for j in offsets {
+                    if columns_info[*j as usize].as_accessor().is_string_like() {
+                        hasher.write(&collation_keys_val[*j as usize]);
+                    } else {
+                        hasher.write(&columns_val[*j as usize]);
+                    }
                 }
+                self.fm_sketches[col_len + i].insert_hash_value(hasher.finish());
```

Comment:
looks a little bit suspicious to me that we are using different ways of computation for len = 1 and len != 1, I don't think it's consistent with the previous way of computation...

Reply:
As [mur3 doc](https://docs.rs/mur3/latest/mur3/) suggests, Hasher version(`Hasher128`) is a little slower than the function version `murmurhash3_x64_128`. Hence, when len = 1, we use the faster one(`murmurhash3_x64_128`), while when len > 1, we use `Hasher128` to avoid concating byte slices.